### ISSUE TITLE
Login error

### ISSUE THREAD

**Issue by ulrith** [maxking/docker-mailman#53]
Hi again, it seems I hit another bug.
The web interface working just fine if I login on page /admin/
But if I logout and then try to login again on page
/accounts/login/?next=/postorius/lists/
using the same credentials I've got error code 500 with the following message in the logs:
```
ERROR 2017-06-15 20:32:23,236 22 django.request Internal Server Error: /accounts/login/
Traceback (most recent call last):
  File "/usr/local/lib/python2.7/site-packages/django/core/handlers/exception.py", line 39, in inner
    response = get_response(request)
  File "/usr/local/lib/python2.7/site-packages/django/core/handlers/base.py", line 249, in _legacy_get_response
    response = self._get_response(request)
  File "/usr/local/lib/python2.7/site-packages/django/core/handlers/base.py", line 187, in _get_response
    response = self.process_exception_by_middleware(e, request)
  File "/usr/local/lib/python2.7/site-packages/django/core/handlers/base.py", line 185, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "/usr/local/lib/python2.7/site-packages/django/views/generic/base.py", line 68, in view
    return self.dispatch(request, *args, **kwargs)
  File "/usr/local/lib/python2.7/site-packages/django/utils/decorators.py", line 67, in _wrapper
    return bound_func(*args, **kwargs)
  File "/usr/local/lib/python2.7/site-packages/django/views/decorators/debug.py", line 76, in sensitive_post_parameters_wrapper
    return view(request, *args, **kwargs)
  File "/usr/local/lib/python2.7/site-packages/django/utils/decorators.py", line 63, in bound_func
    return func.__get__(self, type(self))(*args2, **kwargs2)
  File "/usr/local/lib/python2.7/site-packages/allauth/account/views.py", line 128, in dispatch
    return super(LoginView, self).dispatch(request, *args, **kwargs)
  File "/usr/local/lib/python2.7/site-packages/allauth/account/views.py", line 74, in dispatch
    **kwargs)
  File "/usr/local/lib/python2.7/site-packages/django/views/generic/base.py", line 88, in dispatch
    return handler(request, *args, **kwargs)
  File "/usr/local/lib/python2.7/site-packages/allauth/account/views.py", line 97, in post
    response = self.form_valid(form)
  File "/usr/local/lib/python2.7/site-packages/allauth/account/views.py", line 141, in form_valid
    return form.login(self.request, redirect_url=success_url)
  File "/usr/local/lib/python2.7/site-packages/allauth/account/forms.py", line 177, in login
    redirect_url=redirect_url)
  File "/usr/local/lib/python2.7/site-packages/allauth/account/utils.py", line 149, in perform_login
    send_email_confirmation(request, user, signup=signup)
  File "/usr/local/lib/python2.7/site-packages/allauth/account/utils.py", line 313, in send_email_confirmation
    signup=signup)
  File "/usr/local/lib/python2.7/site-packages/allauth/account/models.py", line 60, in send_confirmation
    confirmation.send(request, signup=signup)
  File "/usr/local/lib/python2.7/site-packages/allauth/account/models.py", line 171, in send
    get_adapter(request).send_confirmation_mail(request, self, signup)
  File "/usr/local/lib/python2.7/site-packages/allauth/account/adapter.py", line 454, in send_confirmation_mail
    ctx)
  File "/usr/local/lib/python2.7/site-packages/allauth/account/adapter.py", line 142, in send_mail
    msg.send()
  File "/usr/local/lib/python2.7/site-packages/django/core/mail/message.py", line 342, in send
    return self.get_connection(fail_silently).send_messages([self])
  File "/usr/local/lib/python2.7/site-packages/django/core/mail/backends/smtp.py", line 107, in send_messages
    sent = self._send(message)
  File "/usr/local/lib/python2.7/site-packages/django/core/mail/backends/smtp.py", line 123, in _send
    self.connection.sendmail(from_email, recipients, message.as_bytes(linesep='\r\n'))
  File "/usr/local/lib/python2.7/smtplib.py", line 747, in sendmail
    raise SMTPRecipientsRefused(senderrs)
SMTPRecipientsRefused: {u'user@email.com': (451, '4.3.0 <user@email.com>: Temporary lookup failure')}

```

**Comment by ulrith** [comment]
And I have many error messages in the logs:
```
Jun 15 23:37:55 host1 postfix/smtpd[9980]: NOQUEUE: reject: RCPT from unknown[172.19.199.18]: 451 4.3.0 <user@email.com>: Temporary lookup failure; from=<test-bounces@a91ce067-8092-467e-93f9-f27ac15ad7ea.mailman3.com> to=<user@email.com> proto=ESMTP helo=<[172.19.199.18]>
```

**Comment by ulrith** [comment]
After some debugging I realized that this error related to the situation when no lists created. In this case /var/data/ folder for mailman-core appears empty (no postfix_* files created yet) which breaks normal work of postfix on the host system.

**Comment by maxking** [comment]
It sounds like a postfix configuration problem to me, it should be configured to relay emails from the web container, which it looks like is not right now.

**Comment by ulrith** [comment]
Yes, as I wrote before, this is because of absence of postfix configuration files in core/var/data folder before the first list creation, which is definitely bug to me.

**Comment by maxking** [comment]
@ulrith no, I was looking at this: 
```Jun 15 23:37:55 host1 postfix/smtpd[9980]: NOQUEUE: reject: RCPT from unknown[172.19.199.18]: 451 4.3.0 <user@email.com>: Temporary lookup failure; from=<test-bounces@a91ce067-8092-467e-93f9-f27ac15ad7ea.mailman3.com> to=<user@email.com> proto=ESMTP helo=<[172.19.199.18]>```
And it looks like it is rejecting emails for this container.

**Comment by danil-smirnov** [comment]
Yes, but a reason why it rejects emails is what I've talked about.
After those files appear (on list creation) the error has gone.

So the conclusion is that if you have references to those files in postfix config, they _must exist_ from the very beginning of the containers lifecycle. Otherwise you **can not login** until first list is created - which is not by design (I hope).

**Comment by ulrith** [comment]
Workaround: just create files right after containers creation, the reload postfix:
```
touch /opt/mailman/core/var/data/postfix_domains
chown --reference=/opt/mailman/core/var/data/ /opt/mailman/core/var/data/postfix_domains
chmod 660 /opt/mailman/core/var/data/postfix_domains

touch /opt/mailman/core/var/data/postfix_lmtp
chown --reference=/opt/mailman/core/var/data/ /opt/mailman/core/var/data/postfix_lmtp
chmod 660 /opt/mailman/core/var/data/postfix_lmtp
```

**Comment by maxking** [comment]
@ulrith You might be able to solve this by just running `docker-compose -T mailman-core mailman aliases`.

**Comment by ulrith** [comment]
The correct command is _docker-compose **exec** -T mailman-core mailman aliases_
and yes - it works. But only after containers were completely up.

You can create files independently, but to exec something inside containers you need to ensure that all post-install tasks like migrations are completed.

@maxking Can you advise some way to check if the containers are completely up?

**Comment by maxking** [comment]
@ulrith Not sure about that, you can maybe check if the endpoints are reachable to detect if the images are up? 

I use this logic to detect if the postgres images are up before I start the mailman or django in the provided images. For now, there is no such thing happening for MySQL, but it probably should be added.

**Comment by maxking** [comment]
You can have a biref look at this stupid tests I have just to make sure that the containers are actually running: https://github.com/maxking/docker-mailman/blob/master/tests/test.sh

Might be of some help.

**Comment by ulrith** [comment]
@maxking I'm doing similar thing - grepping containters logs waiting for some string. But not the best approaches though

**Comment by maxking** [comment]
Yeah, but I am not sure how to actually provide that information.

**Comment by ulrith** [comment]
The best way would be to have /healthcheck endpoints for all apps which become available after all post-install tasks have succeed...

**Comment by maxking** [comment]
But that isn't something which can be done as a part of packaging.

**Comment by almereyda** [comment]
Due to the `WORKDIR` of the `mailman-core` Image/Dockerfile, we had to use

    docker-compose exec mailman-core sh -c "cd core && mailman aliases"

**Comment by maxking** [comment]
@almereyda So, if mailman can't find the configuration file at locations it searches and the environment variable MAILMAN_CONFIG_FILE isn't set, it usually creates a new `var` directory with basic default configuration and new sqlite database and everything else like postfix_lmtp maps.

This *should* not happen in the containers because they have MAILMAN_CONFIG_FILE env variable set and pointing to `/etc/mailman.cfg` which would prevent the creating of `var` dir in `pwd`.

I will have to check once to make sure that this doesn't happen.

If postfix complains about missing files at startup, it might be worth opening an issue at https://gitlab.com/mailman/mailman/issues to make sure these files are generated at startup time and don't require running additional commands. It would be a more appropriate place to discuss if it should be a part of packaging or Core itself.

**Comment by maxking** [comment]
Fixed by #77
